tests/pyoozie/test_builder.py

```diff
@@ -130,82 +118,30 @@ def test_coordinator_submission_xml_with_configuration(username, coord_app_path)
-            error=e.output.decode('utf8').strip(),
-            xml=actual_xml,
-        ))
-
-
-def test_builder_raises_on_bad_workflow_name():
-    # Does it throw an exception on a bad workflow name?
-    with pytest.raises(AssertionError) as assertion_info:
-        builder.WorkflowBuilder(
-            name='l' * (tags.MAX_NAME_LENGTH + 1)
-        ).add_action(
-            name='payload',
-            action=tags.Shell(exec_command='echo "test"'),
-            action_on_error=tags.Email(to='person@example.com', subject='Error', body='A bad thing happened'),
-            kill_on_error='Failure message',
-        )
-    assert "Name must be less than " in str(assertion_info.value)
-
-
-def test_builder_raises_on_bad_action_name():
-    # Does it throw an exception on a bad action name?
-    with pytest.raises(AssertionError) as assertion_info:
-        builder.WorkflowBuilder(
-            name='descriptive-name'
-        ).add_action(
-            name='Action name with invalid characters',
-            action=tags.Shell(exec_command='echo "test"'),
-            action_on_error=tags.Email(to='person@example.com', subject='Error', body='A bad thing happened'),
-            kill_on_error='Failure message',
-        )
-    assert "Identifier must match " in str(assertion_info.value) and \
-        "Action name with invalid characters" in str(assertion_info.value)
-
-    # Does it throw an exception on an action name that's too long?
-    with pytest.raises(AssertionError) as assertion_info:
-        builder.WorkflowBuilder(
-            name='descriptive-name'
-        ).add_action(
-            name='l' * (tags.MAX_IDENTIFIER_LENGTH + 1),
-            action=tags.Shell(exec_command='echo "test"'),
-            action_on_error=tags.Email(to='person@example.com', subject='Error', body='A bad thing happened'),
-            kill_on_error='Failure message',
-        )
-    assert "Identifier must be less than " in str(assertion_info.value)
-
-
-def test_builder_raises_on_multiple_actions(workflow_builder):
-    # Does it raise an exception when you try to add multiple actions?
-    with pytest.raises(NotImplementedError) as assertion_info:
-        workflow_builder.add_action(
-            name='payload',
-            action=tags.Shell(exec_command='echo "test"'),
-            action_on_error=tags.Email(to='person@example.com', subject='Error', body='A bad thing happened'),
-            kill_on_error='Failure message',
-        )
-    assert str(assertion_info.value) == 'Can only add one action in this version'
+@pytest.mark.xfail(raises=NotImplementedError)
+def test_workflow_builder():
+
+    workflow_builder = builder.WorkflowBuilder(
+        name='descriptive-name',
+        actions=workflow.Sequence(
+            workflow.Action(tags.Shell(exec_command='echo', arguments=("'A'",))),
+            workflow.Parallelization(
+                workflow.Sequence(
```

Comment:
> In my mental model all three should be equivalent

Correct.

`Parallelization` should take either another `Parallelization` (i.e. nested fork/join), a concrete `Action` (`Shell`, `Email`, `SubWorkflow`), or `Sequence`.
`Serial` should also accept another `Parallelization` or `Sequence` (allowing a sequence to share the same error handler).